qiskit_nature/circuit/library/initial_states/_helper.py

```diff
@@ -0,0 +1,197 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021, 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Private helper functions for initial states."""
+
+from typing import Iterable, List, Sequence, Tuple, Union
+
+import numpy as np
+
+from qiskit import QuantumRegister
+from qiskit.circuit import Gate, Qubit
+from qiskit.circuit.library import RZGate, XGate, XXPlusYYGate
+
+
+def _prepare_slater_determinant_jordan_wigner(  # pylint: disable=invalid-name
```

Comment:
We have moved away from hard-coded fermion-to-qubit transformations everywhere in the stack. There are rare exceptions where a certain mapping is not allowed but in general it should not be the case that only one single case is allowed.

If there is no way of implementing a general method which works for at least JW and parity, please explain why.

Reply:
>There are rare exceptions where a certain mapping is not allowed but in general it should not be the case that only one single case is allowed.

Many algorithms in the literature (such as this one) are designed specifically for the Jordan-Wigner transform, so this does not seem like a tenable policy. It may be possible to adapt them to the parity transform, but I think the benefit hardly justifies the effort, since the parity transform has no advantages over the Jordan-Wigner transform. It seems unlikely that these algorithms could be adapted to the Bravyi-Kitaev transform.

Reply:
Well the parity transform has the benefit of allowing a trivial reduction of two qubits.

I agree that the BK transform may be out-of-scope which, as I said above, is also the case elsewhere.

Reply:
Okay, we can leave it as is. However, please add unittest to ensure that non-JW mappers are caught gracefully :+1:

Reply:
Done.